src/app.py

```diff
@@ -154,6 +185,8 @@ def write_required(f):
     def decorated_function(*args, **kwargs):
         user = current_user
         if not user.has_write_permissions():
+            current_app.logger.debug(
```

Comment:
I'd make this higher than debug since you don't expect it to happen normally.

Either there's a bug or someone is manually attempting a request they're not authorised to make